src/services/merkle-car-service.ts

```diff
@@ -0,0 +1,56 @@
+import { CARFactory, CAR } from 'cartonne'
+import type { CID } from 'multiformats/cid'
+import AWSSDK from 'aws-sdk'
+import LevelUp from 'levelup'
+import S3LevelDOWN from 's3leveldown'
+import * as DAG_JOSE from 'dag-jose'
+import { logger } from '../logger/index.js'
+
+/**
+ * A service for storing and retrieving the CAR file containing the entire anchor merkle tree,
+ * along with the associated anchor proof and anchor commits.
+ */
+export interface IMerkleCarService {
+  storeCarFile(anchorProofCID: CID, car: CAR): Promise<void>
+  retrieveCarFile(anchorProofCID: CID): Promise<CAR | null>
+}
+
+export class InMemoryMerkleCarService {
+  readonly cars: Map<string, CAR> = new Map()
+
+  async storeCarFile(anchorProofCID: CID, car: CAR): Promise<void> {
+    this.cars.set(anchorProofCID.toString(), car)
+  }
+
+  async retrieveCarFile(anchorProofCID: CID): Promise<CAR | null> {
+    return this.cars.get(anchorProofCID.toString()) || null
+  }
+}
+
+const S3_STORE_PATH = '/ceramic/cas/anchor/merkle-car/'
+const carFactory = new CARFactory()
+carFactory.codecs.add(DAG_JOSE)
+
+export class S3MerkleCarService {
+  readonly s3store: LevelUp.LevelUp = new LevelUp(new S3LevelDOWN(S3_STORE_PATH, new AWSSDK.S3()))
```

Comment:
How is it configured? I mean, I would expect some credentials and a name of s3 bucket. Does that happens automagically somehow?

Reply:
Yeah, my understanding is that the S3 library knows to look for certain specific environment variables and gets its configuration from that.  I believe that's how we usually configure the S3 settings for our Ceramic nodes currently.  @smrz2001 @3benbox can you confirm?